src/slack/slack_types.bal

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/oauth2;
+import ballerina/http;
+
+# Contains information about the channel topic.
+# 
+# + value - Value of the topic
+# + creator - The user Id of the member that created the topic
+public type Topic record {
```

Comment:
Why all these records are `Open records`? Can the user add more fields and make use of it?

Reply:
These records are mapped to the json responses coming from the slack API. We haven't added each and every field of that json response in the record. Only a chosen set of fields are added. Hence we need to keep them as open records.